src/stream/src/executor/top_n/topn_cache_state.rs

```diff
@@ -108,18 +108,25 @@ impl TopNCacheState {
         self.inner.range(range)
     }
 
-    pub fn extract_if<F>(&mut self, pred: F) -> ExtractIf<'_, CacheKey, CompactedRow, F, Global>
+    pub fn extract_if<'a, F1>(
+        &'a mut self,
+        mut pred: F1,
```

Comment:
Why `mut` here? And no need to change to `F1`

Reply:
Just learn form @BugenZhao, FnMut::call_mut accept &mut self. Calling `pred` will mutate itself.